yt/visualization/plot_modifications.py

```diff
@@ -1254,44 +1273,41 @@ def __init__(
         self.plot_args = plot_args
 
     def __call__(self, plot):
-        bounds = self._physical_bounds(plot)
         xx0, xx1, yy0, yy1 = self._plot_bounds(plot)
 
         # We are feeding this size into the pixelizer, where it will properly
         # set it in reverse order
         nx = plot.raw_image_shape[1] // self.factor[0]
         ny = plot.raw_image_shape[0] // self.factor[1]
-        pixX = plot.data.ds.coordinates.pixelize(
-            plot.data.axis, plot.data, self.field_x, bounds, (nx, ny)
-        )
-        pixY = plot.data.ds.coordinates.pixelize(
-            plot.data.axis, plot.data, self.field_y, bounds, (nx, ny)
+        pixelize = partial(
+            plot.data.ds.coordinates.pixelize,
+            plot.data.axis,
+            plot.data,
+            bounds=self._physical_bounds(plot),
+            size=(nx, ny),
         )
-        if self.field_color:
-            field_colors = plot.data.ds.coordinates.pixelize(
-                plot.data.axis, plot.data, self.field_color, bounds, (nx, ny)
-            )
-
-            if self.display_threshold:
-                mask = field_colors > self.display_threshold
-                lwdefault = matplotlib.rcParams["lines.linewidth"]
+        pixX = pixelize(field=self.field_x)
+        pixY = pixelize(field=self.field_y)
+        if (
+            isinstance(self._linewidth, tuple)
+            and len(self._linewidth) == 2
+            and all(isinstance(_, str) for _ in self._linewidth)
+        ):
+            linewidth = pixelize(field=self._linewidth)
+            if plot._swap_axes:
+                linewidth = linewidth.transpose()
+        else:
+            linewidth = self._linewidth
 
-                if "linewidth" in self.plot_args:
-                    linewidth = self.plot_args["linewidth"]
-                else:
-                    linewidth = lwdefault
-
-                try:
-                    linewidth *= mask
-                    self.plot_args["linewidth"] = linewidth
-                except ValueError as e:
-                    err_msg = (
-                        "Error applying display threshold: linewidth"
-                        + "must have shape ({}, {}) or be scalar"
-                    )
-                    err_msg = err_msg.format(nx, ny)
-                    raise ValueError(err_msg) from e
+        if linewidth is not None:
+            linewidth *= self._linewidth_upscaling / np.abs(linewidth).max()
```

Comment:
I should probably expose the normalisation factor directly and document that leaving it unspecified will result in normalising by the field's maximum.

Reply:
Not sure.  Does it know how to normalize on log-scaled fields?

Reply:
It doesn't, but there's actually already a way to use *arbitrary* scaling here by defining custom fields that can then be used as `linewidth`, so in hindsight I think it's already general enough. I could explicitly mention this idea in docs.

Reply:
Maybe it should indeed take logs automatically when required by field_info. TBH I don't think much of this mechanism because I feel like it locks users into arbitrary defaults instead of empowering them, but consistency should prevail, especially since, as I pointed out, there's already a way to get out of defaults completely.
That said, and while we're at it, is there a way to (re)set "take_log" parameters from the user side that I just don't know about ?

Reply:
I think it's OK to go in as is, and I think improving field definitions to more flexibly support logs is a good medium-term goal.